### DISCUSSION TITLE
[node] - Node Team are having a discussion about end-user types

### DISCUSSION THREAD

**Discussion by orta** [DefinitelyTyped/DefinitelyTyped#56953] [Issues with a @types package]
Hi people, I figured this was a good way to tell folks who care about the types for Node. It looks like the Node team are thinking of trying to ship first party types and I bet folks who have been maintaining DT's node types would probably have a lot of nuanced opinions which are worth hearing from (or are maybe interested in helping out).

You can see the details here  https://github.com/nodejs/next-10/issues/99

/cc @bnb

[a comment by an automated account is not shown]

**Comment by mohsen1** [comment]
The docs and types must be all autogenerated from code I think. It will be _a lot_ of work!

**Reply by bnb** [reply]
Electron does .d.ts definitions generated from docs with [docs-parser](https://github.com/electron/docs-parser) which enforces our [styleguide](https://github.com/electron/electron/blob/main/docs/styleguide.md) as a way to extract types from docs.

This system is more specifically tuned to their API (most notably, single export per module which is not how Node.js is architected), but I'd really love to see an adaption/rewrite of the underlying tooling that could support both use cases with a minimum criteria of supporting Node.js's use case to accomplish the same goal.

**Comment by bnb** [comment]
Would really love to see those who've got an interest/stake in the DT node types in that session/async <3

**Comment by SimonSchick** [comment]
Mini rant:
The current NodeJS docs are of insufficient detail and quality to generate types of the same quality as we have right now, trust me, I've tried.

I am generally in favor of the NodeJS team managing those types as it will ensure atomicity and consistency of releasing types w/ node and reduce overall churn.

There are quite a few interesting challenges to overcome here and I'd be interesting in attending this.

**Reply by orta** [reply]
This is now BTW Simon - https://github.com/nodejs/next-10/issues/99
